From 598484767fb67d4352662e9bf2fad34b5d8d5be6 Mon Sep 17 00:00:00 2001
From: Mykola Solianko <mykola_solianko@epam.com>
Date: Tue, 23 Apr 2024 08:49:32 +0300
Subject: [PATCH] [network] Add SO_REUSEPORT to allow tcp listening

Signed-off-by: Mykola Solianko <mykola_solianko@epam.com>
---
 src/network.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network.c b/src/network.c
index b061833..00c860e 100644
--- a/src/network.c
+++ b/src/network.c
@@ -942,7 +942,7 @@ static int make_sock(union mysockaddr *addr, int type, int dienow)
       return -1;
     }	
   
-  if (setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &opt, sizeof(opt)) == -1 || !fix_fd(fd))
+  if (setsockopt(fd, SOL_SOCKET, SO_REUSEADDR | SO_REUSEPORT, &opt, sizeof(opt)) == -1 || !fix_fd(fd))
     goto err;
   
   if (family == AF_INET6 && setsockopt(fd, IPPROTO_IPV6, IPV6_V6ONLY, &opt, sizeof(opt)) == -1)
-- 
2.34.1

